### PR TITLE
Patch editor

### PR DESCRIPTION
**Pull request by laerreal**
A tool for editing patches represented by files.
Main purpose is patch splitting and moving hunks between patches.
It supports patches in `git diff` output format.
There also is a front-end for editing/addition commit information for `git am`.

### v5
- forgotten `object`
- coding style

### v4
- fixup commas

### v3
- Patch editor
    - fixup/unify coding style
    - fixup Py2 support

### v2
- fixup misprints
- fixup `__all__`'s coding style

### CODE REVIEW

File: misc/patch_edit.py [hunk 1]
```diff
@@ -0,0 +1,1376 @@
...
+        description = "Patch Editor",
+    )
+    ap.add_argument("patch",
+        nargs = "*",
+    )
+
+    args = ap.parse_args()
+
+    root = PatchEditor()
+    root.patch_file_names = args.patch
```
Comment (BDanAnd) [inline misc/patch_edit.py:1384]: Does not work on Py2.

File: misc/patch_edit.py [hunk 2]
```diff
@@ -0,0 +1,1376 @@
...
+                popup = self._file_popup
+            else:
+                popup = self._dir_popup
+
+            self.show_popup(e.x_root, e.y_root, popup,
+                tag = _id
+            )
+
+    def select_file(self, i):
+        if  hasattr(self, "_reading_task"):
```
Comment (BDanAnd) [inline misc/patch_edit.py]: two spaces

**Review by BDanAnd — COMMENTED** [review]
v1 viewed.

**Comment by laerreal** [comment]
v3 is ready

File: misc/patch_edit.py [hunk 3]
```diff
@@ -0,0 +1,1383 @@
...
+    def commit_info(self):
+        return self._commit_info
+
+    @commit_info.setter
+    def commit_info(self, commit_info):
+        self._commit_info = commit_info
+        self._on_revert()
+
+
+class PatchSeriesEditorFrame(GUIFrame, TkPopupHelper):
```
Comment (BDanAnd) [inline misc/patch_edit.py]: inheritance of `object` is also required for Py2

**Review by BDanAnd — COMMENTED** [review]
v4 viewed.

**Comment by laerreal** [comment]
v5 is ready

**Pull request merged**